### PR TITLE
feat: add linter

### PR DESCRIPTION
**Pull request by matejaVuradin**
-add ESLint and Prettier

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v18
\ No newline at end of file
+18
```
Comment (BeroBurny) [inline .nvmrc]: Can you revert this to be back `v18` as is been orginally??

File: .eslintrc.cjs [hunk 2]
```diff
@@ -1,16 +1,5 @@
+require("@rushstack/eslint-patch/modern-module-resolution");
+
```
Comment (BeroBurny) [inline .eslintrc.cjs]: Think this package `@rushstack/eslint-patch/modern-module-resolution` is unnecessary and can be removed

File: package.json [hunk 3]
```diff
@@ -44,13 +51,17 @@
     "@types/node": "^18",
     "@types/react": "^18.0.37",
     "@types/react-dom": "^18.0.11",
-    "@typescript-eslint/eslint-plugin": "^5.59.0",
-    "@typescript-eslint/parser": "^5.59.0",
+    "@typescript-eslint/eslint-plugin": "latest",
+    "@typescript-eslint/parser": "latest",
```
Comment (BeroBurny) [inline package.json]: semantic versioning is preferred over the `latest`

**Review by BeroBurny — CHANGES_REQUESTED** [review]

File: .nvmrc (hunk 1 above)
Comment (matejaVuradin) [reply .nvmrc]: done

File: .eslintrc.cjs (hunk 2 above)
Comment (matejaVuradin) [reply .eslintrc.cjs]: done

File: package.json (hunk 3 above)
Comment (matejaVuradin) [reply package.json]: done

**Review by BeroBurny — APPROVED** [review]
LGTM

**Pull request merged**